data/types/base/blood/glucose/glucose_test.go

```diff
@@ -0,0 +1,223 @@
+package glucose_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"math"
+
+	"github.com/tidepool-org/platform/app"
+	"github.com/tidepool-org/platform/data/context"
+	"github.com/tidepool-org/platform/data/normalizer"
+	"github.com/tidepool-org/platform/data/types/base"
+	"github.com/tidepool-org/platform/data/types/base/blood/glucose"
+	"github.com/tidepool-org/platform/data/types/base/testing"
+	"github.com/tidepool-org/platform/data/validator"
+	"github.com/tidepool-org/platform/log"
+	"github.com/tidepool-org/platform/service"
+)
+
+func NewMeta() interface{} {
+	return &base.Meta{
+		Type: "testGlucose",
+	}
+}
+
+func NewTestGlucose(sourceTime interface{}, sourceUnits interface{}, sourceValue interface{}) *glucose.Glucose {
+	testGlucose := &glucose.Glucose{}
+	testGlucose.Init()
+	testGlucose.Type = "testGlucose"
+	testGlucose.DeviceID = app.StringAsPointer(app.NewID())
+	if value, ok := sourceTime.(string); ok {
+		testGlucose.Time = app.StringAsPointer(value)
+	}
+	if value, ok := sourceUnits.(string); ok {
+		testGlucose.Units = app.StringAsPointer(value)
+	}
+	if value, ok := sourceValue.(float64); ok {
+		testGlucose.Value = app.FloatAsPointer(value)
+	}
+	return testGlucose
+}
+
+var _ = Describe("Glucose", func() {
+	Context("with new glucose", func() {
+		var testGlucose *glucose.Glucose
+
+		BeforeEach(func() {
+			testGlucose = &glucose.Glucose{}
+		})
+
+		Context("with initialized", func() {
+			BeforeEach(func() {
+				testGlucose.Init()
+			})
+
+			DescribeTable("Validate",
+				func(sourceGlucose *glucose.Glucose, expectedErrors []*service.Error) {
+					testContext, err := context.NewStandard(log.NewNull())
+					Expect(err).ToNot(HaveOccurred())
+					Expect(testContext).ToNot(BeNil())
+					testValidator, err := validator.NewStandard(testContext)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(testValidator).ToNot(BeNil())
+					Expect(sourceGlucose.Validate(testValidator)).To(Succeed())
+					Expect(testContext.Errors()).To(ConsistOf(expectedErrors))
+				},
+				Entry("all valid",
+					NewTestGlucose("2016-09-06T13:45:58-07:00", "mmol/L", 10.0),
+					[]*service.Error{}),
+				Entry("missing time",
+					NewTestGlucose(nil, "mmol/L", 10.0),
+					[]*service.Error{
+						testing.ComposeError(service.ErrorValueNotExists(), "/time", NewMeta()),
+					}),
+				Entry("missing units",
+					NewTestGlucose("2016-09-06T13:45:58-07:00", nil, 10.0),
+					[]*service.Error{
+						testing.ComposeError(service.ErrorValueNotExists(), "/units", NewMeta()),
+					}),
+				Entry("unknown units",
+					NewTestGlucose("2016-09-06T13:45:58-07:00", "unknown", 10.0),
+					[]*service.Error{
+						testing.ComposeError(service.ErrorValueStringNotOneOf("unknown", []string{"mmol/L", "mmol/l", "mg/dL", "mg/dl"}), "/units", NewMeta()),
```

Comment:
If we test using the constants then what is to prevent someone from accidentally changing the constants?